src/IO/AsynchronousReadBufferFromFileDescriptor.cpp

```diff
@@ -109,7 +112,10 @@ bool AsynchronousReadBufferFromFileDescriptor::nextImpl()
         auto [size, offset] = asyncReadInto(memory.data(), memory.size()).get();
         file_offset_of_buffer_end += size;
 
-        if (size)
```

Comment:
Previously, if we have read data with offset till the end of file, we returned `true` for `nextImpl` call, which could lead to mistakenly returning `false` from `eof()` and unexpected reading after `eof`.